c/src/ml-api-remote-service.c

```diff
@@ -562,28 +584,55 @@ ml_remote_service_register (ml_service_h handle, ml_option_h option, void *data,
     return ret;
   }
 
-  nns_edge_data_set_info (data_h, "service-type", service_str);
-  nns_edge_data_set_info (data_h, "service-key", service_key);
-  ml_option_get (option, "description", (void **) &description);
-  nns_edge_data_set_info (data_h, "description", description);
-  ml_option_get (option, "name", (void **) &name);
-  nns_edge_data_set_info (data_h, "name", name);
-  ml_option_get (option, "activate", (void **) &activate);
-  nns_edge_data_set_info (data_h, "activate", activate);
+  ret = nns_edge_data_set_info (data_h, "service-type", service_str);
+  if (NNS_EDGE_ERROR_NONE != ret) {
+    _ml_error_report ("Failed to set service type in edge data.");
+    goto done;
+  }
+  ret = nns_edge_data_set_info (data_h, "service-key", service_key);
+  if (NNS_EDGE_ERROR_NONE != ret) {
+    _ml_error_report ("Failed to set service key in edge data.");
+    goto done;
+  }
+  ret = ml_option_get (option, "description", (void **) &description);
+  if (ML_ERROR_NONE != ret) {
+    _ml_logi ("Failed to get option description.");
+  }
```

Comment:
Does it need ```goto done``` ?

Reply:
It's not necessary option, so just inform to the user.